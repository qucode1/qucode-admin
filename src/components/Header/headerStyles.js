export const headerStyles = `
.header {
  background-image: linear-gradient(to right, #22bbee, #0033ee);
  color: white;
  position: fixed;
  top: 0;
  width: 100%;
  height: 70px
}
.container {
  max-width: var(--site-width);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 10px;
  margin: 0 auto;
}
.logo {
  font-style: italic;
}
.nav {
  display: flex;
  justify-content: center;
}
.navItem {
  font-weight: bold;
  text-transform: uppercase;
  margin: 0 0 0 15px;
  font-size: 2rem
}
.link {
  color: #fff;
}
.link:hover, .link:focus {
  color: #33ceff
}
`
