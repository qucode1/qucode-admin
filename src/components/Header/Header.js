import React from 'react'
import { headerStyles } from './headerStyles'
import { Link } from 'react-router-dom'

const Header = () => (
  <header className='header'>
    <div className='container'>
      <h1 className='logo'><Link to='/' className='link'>QuCode | Admin Panel</Link></h1>
      <nav>
        <ul className='nav'>
          <li className='navItem'><Link className='link' to='/posts'>Posts</Link></li>
          <li className='navItem'><Link className='link' to='/about'>About</Link></li>
        </ul>
      </nav>
    </div>
    <style jsx>{ headerStyles }</style>
  </header>
)
export default Header
