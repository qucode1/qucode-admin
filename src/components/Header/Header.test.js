import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Header from './Header'

describe('<Header />', () => {
  it('should render title and nav as expected', () => {
    const wrapper = shallow(<Header />)
    const tree = toJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
})
