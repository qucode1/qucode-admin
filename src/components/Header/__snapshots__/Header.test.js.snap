// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render title and nav as expected 1`] = `
<header
  className="heading"
>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      <Link
        className="link"
        replace={false}
        to="/"
      >
        Aletta
      </Link>
    </h1>
    <nav>
      <ul
        className="list"
      >
        <li
          className="listItem"
        >
          <Link
            className="link"
            replace={false}
            to="/posts"
          >
            Posts
          </Link>
        </li>
        <li
          className="listItem"
        >
          <Link
            className="link"
            replace={false}
            to="/about"
          >
            About
          </Link>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
